### ISSUE TITLE
Window image not showing, with no error on MacOS Mojave

### ISSUE THREAD

**Issue by ianshmean** [JuliaImages/ImageView.jl#175]
On a fresh julia environment on MacOS Mojave I get the following, without error messages:

![image](https://user-images.githubusercontent.com/1694067/56996439-8f87db80-6b72-11e9-838d-0fe29522081b.png)

I don't see this issue on Ubuntu 18.04

**Comment by timholy** [comment]
Huh. Does `Gtk.showall(dct["gui"]["window"])` do anything for you? (`dct` being the output of `imshow`) If not, do Gtk and GtkReactive work for you in general? (Try their tests.)

**Comment by ianshmean** [comment]
`Gtk.showall(dct["gui"]["window"])` produces the following but still the window is blank
```julia
julia> Gtk.showall(dct["gui"]["window"])
GtkWindowLeaf(name="", parent, width-request=-1, height-request=-1, visible=TRUE, sensitive=TRUE, app-paintable=FALSE, can-focus=FALSE, has-focus=FALSE, is-focus=FALSE, focus-on-click=TRUE, can-default=FALSE, has-default=FALSE, receives-default=FALSE, composite-child=FALSE, style, events=0, no-show-all=FALSE, has-tooltip=FALSE, tooltip-markup=NULL, tooltip-text=NULL, window, opacity=1.000000, double-buffered, halign=GTK_ALIGN_FILL, valign=GTK_ALIGN_FILL, margin-left, margin-right, margin-start=0, margin-end=0, margin-top=0, margin-bottom=0, margin=0, hexpand=FALSE, vexpand=FALSE, hexpand-set=FALSE, vexpand-set=FALSE, expand=FALSE, scale-factor=2, border-width=0, resize-mode, child, type=GTK_WINDOW_TOPLEVEL, title="ImageView", role=NULL, resizable=TRUE, modal=FALSE, window-position=GTK_WIN_POS_NONE, default-width=512, default-height=512, destroy-with-parent=FALSE, hide-titlebar-when-maximized=FALSE, icon, icon-name=NULL, screen, type-hint=GDK_WINDOW_TYPE_HINT_NORMAL, skip-taskbar-hint=FALSE, skip-pager-hint=FALSE, urgency-hint=FALSE, accept-focus=TRUE, focus-on-map=TRUE, decorated=TRUE, deletable=TRUE, gravity=GDK_GRAVITY_NORTH_WEST, transient-for, attached-to, has-resize-grip, resize-grip-visible, application, is-active=FALSE, has-toplevel-focus=FALSE, startup-id, mnemonics-visible=FALSE, focus-visible=FALSE, is-maximized=FALSE)
```

`] test Gtk` passes but each output is just a blank window

`] test GtkReactive` fails!

```julia
(v1.1) pkg> test GtkReactive
   Testing GtkReactive
 Resolving package versions...
 Installed ImageMagick ─ v0.7.3
  Building ImageMagick → `~/tmp/pkgs/packages/ImageMagick/J74FC/deps/build.log`
    Status `/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/tmpnOvPVb/Manifest.toml`
  [621f4979] AbstractFFTs v0.4.0
  [39de3d68] AxisArrays v0.3.0
  [9e28174c] BinDeps v0.8.10
  [b99e7846] BinaryProvider v0.5.4
  [e1450e63] BufferedStreams v1.0.0
  [159f3aea] Cairo v0.6.0
  [3da002f7] ColorTypes v0.7.5
  [c3611d14] ColorVectorSpace v0.7.0
  [5ae59095] Colors v0.9.5
  [34da2185] Compat v2.1.0
  [8f4d0f93] Conda v1.2.0
  [864edb3b] DataStructures v0.15.0
  [7a1cc6ca] FFTW v0.2.4
  [5789e2e9] FileIO v1.0.6
  [53c48c17] FixedPointNumbers v0.5.3
  [a2bd30eb] Graphics v0.4.0
  [4c0ca9eb] Gtk v0.17.0
  [27996c0f] GtkReactive v0.6.0
  [0862f596] HTTPClient v0.2.1
  [d9be37ee] Homebrew v0.7.1
  [bbac6d45] IdentityRanges v0.3.0
  [a09fc81d] ImageCore v0.8.2
  [6218d12a] ImageMagick v0.7.3
  [8197267c] IntervalSets v0.3.1
  [c8e1da08] IterTools v1.1.1
  [682c06a0] JSON v0.20.0
  [b27032c2] LibCURL v0.5.0
  [522f3ed2] LibExpat v0.5.0
  [2ec943e9] Libz v1.0.0
  [dbb5928d] MappedArrays v0.2.1
  [e1d29d7a] Missings v0.4.0
  [77ba4419] NaNMath v0.3.2
  [6fe1bfb0] OffsetArrays v0.10.0
  [bac558e1] OrderedCollections v1.1.0
  [5432bcbf] PaddedViews v0.4.2
  [dca85d43] QuartzImageIO v0.6.0
  [b3c3ace0] RangeArrays v0.3.1
  [a223df75] Reactive v0.8.3
  [189a3867] Reexport v0.2.0
  [d5f540fe] RoundingIntegers v0.2.0
  [a2af1166] SortingAlgorithms v0.3.1
  [276daf66] SpecialFunctions v0.7.2
  [2913bbd2] StatsBase v0.30.0
  [5e47fb64] TestImages v0.5.0
  [30578b45] URIParser v0.4.0
  [81def892] VersionParsing v1.1.3
  [c17dfb99] WinRPM v0.4.2
  [a5390f91] ZipFile v0.8.1
  [2a0f44e3] Base64  [`@stdlib/Base64`]
  [ade2ca70] Dates  [`@stdlib/Dates`]
  [8bb1440f] DelimitedFiles  [`@stdlib/DelimitedFiles`]
  [8ba89e20] Distributed  [`@stdlib/Distributed`]
  [b77e0a4c] InteractiveUtils  [`@stdlib/InteractiveUtils`]
  [76f85450] LibGit2  [`@stdlib/LibGit2`]
  [8f399da3] Libdl  [`@stdlib/Libdl`]
  [37e2e46d] LinearAlgebra  [`@stdlib/LinearAlgebra`]
  [56ddb016] Logging  [`@stdlib/Logging`]
  [d6f4376e] Markdown  [`@stdlib/Markdown`]
  [a63ad114] Mmap  [`@stdlib/Mmap`]
  [44cfe95a] Pkg  [`@stdlib/Pkg`]
  [de0858da] Printf  [`@stdlib/Printf`]
  [3fa0cd96] REPL  [`@stdlib/REPL`]
  [9a3f8284] Random  [`@stdlib/Random`]
  [ea8e919c] SHA  [`@stdlib/SHA`]
  [9e88b42a] Serialization  [`@stdlib/Serialization`]
  [1a1011a3] SharedArrays  [`@stdlib/SharedArrays`]
  [6462fe0b] Sockets  [`@stdlib/Sockets`]
  [2f01184e] SparseArrays  [`@stdlib/SparseArrays`]
  [10745b16] Statistics  [`@stdlib/Statistics`]
  [8dfed614] Test  [`@stdlib/Test`]
  [cf7118a7] UUIDs  [`@stdlib/UUIDs`]
  [4ec0a83e] Unicode  [`@stdlib/Unicode`]
Test Summary: | Pass  Total
Widgets       |   65     65
Test Summary:    | Pass  Total
Compound widgets |    3      3
Test Summary: | Pass  Total
CairoUnits    |   15     15
Canvas: Test Failed at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:358
  Expression: Graphics.width(c) == 208
   Evaluated: 1 == 208
Stacktrace:
 [1] top-level scope at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:358
 [2] top-level scope at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Test/src/Test.jl:1083
 [3] top-level scope at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:341
Canvas: Test Failed at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:359
  Expression: Graphics.height(c) == 207
   Evaluated: 1 == 207
Stacktrace:
 [1] top-level scope at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:359
 [2] top-level scope at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Test/src/Test.jl:1083
 [3] top-level scope at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:341
┌ Error: GtkCanvas not yet initialized.
└ @ Gtk ~/tmp/pkgs/packages/Gtk/aP55V/src/cairo.jl:123
Canvas: Error During Test at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:340
  Got exception outside of a @test
  MethodError: no method matching set_coordinates(::Gtk.GLib.FieldRef{Gtk.GtkCanvas}, ::BoundingBox, ::BoundingBox)
  Closest candidates are:
    set_coordinates(!Matched::GraphicsContext, ::BoundingBox, ::BoundingBox) at /Users/ian/tmp/pkgs/packages/Graphics/XgkW4/src/Graphics.jl:381
    set_coordinates(!Matched::GraphicsContext, ::Any, ::Any, !Matched::Any, !Matched::Any, !Matched::Any, !Matched::Any, !Matched::Any, !Matched::Any) at /Users/ian/tmp/pkgs/packages/Graphics/XgkW4/src/Graphics.jl:350
    set_coordinates(!Matched::Union{GtkCanvas, Canvas}, ::BoundingBox, ::BoundingBox) at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/src/GtkReactive.jl:95
    ...
  Stacktrace:
   [1] set_coordinates(::GtkReactive.Canvas{UserUnit}, ::BoundingBox, ::BoundingBox) at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/src/GtkReactive.jl:95
   [2] set_coordinates(::GtkReactive.Canvas{UserUnit}, ::BoundingBox) at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/src/GtkReactive.jl:97
   [3] top-level scope at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:372
   [4] top-level scope at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Test/src/Test.jl:1083
   [5] top-level scope at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:341
   [6] include at ./boot.jl:326 [inlined]
   [7] include_relative(::Module, ::String) at ./loading.jl:1038
   [8] include(::Module, ::String) at ./sysimg.jl:29
   [9] include(::String) at ./client.jl:403
   [10] top-level scope at none:0
   [11] eval(::Module, ::Any) at ./boot.jl:328
   [12] exec_options(::Base.JLOptions) at ./client.jl:243
   [13] _start() at ./client.jl:436
Test Summary: | Pass  Fail  Error  Total
Canvas        |   12     2      1     15
ERROR: LoadError: Some tests did not pass: 12 passed, 2 failed, 1 errored, 0 broken.
in expression starting at /Users/ian/tmp/pkgs/packages/GtkReactive/2h7NX/test/runtests.jl:340
ERROR: Package GtkReactive errored during testing
```

**Comment by damiansp** [comment]
I am having a similar problem, also with Mojave (10.14.3)
`] test Gtk` also passes all tests, but fails to display anything but blank windows, along with:
```
The following messages:
   WARNING: Error in @guarded callback
are expected and a sign of normal operation.
┌ Warning: Error in @guarded callback
└ @ Main.GtkTestModule ~/.julia/packages/Gtk/aP55V/src/base.jl:95
ERROR: UndefVarError: k not defined
┌ Warning: Error in @guarded callback
└ @ Main.GtkTestModule ~/.julia/packages/Gtk/aP55V/src/base.jl:95
ERROR: UndefVarError: k not defined
┌ Warning: Error in @guarded callback
└ @ Main.GtkTestModule ~/.julia/packages/Gtk/aP55V/src/base.jl:95
ERROR: UndefVarError: k not defined
┌ Warning: Error in @guarded callback
└ @ Main.GtkTestModule ~/.julia/packages/Gtk/aP55V/src/base.jl:95
ERROR: UndefVarError: k not defined
Test Summary: | Pass  Total
misc          |    9      9
Test Summary: | Pass  Total
text          |   13     13
   Testing Gtk tests passed

**Comment by timholy** [comment]
The parent issue is https://github.com/JuliaGraphics/Gtk.jl/issues/414. I don't know if it's an OSX version issue, but I'd recommend checking out some older releases of Gtk.jl just to test.

**Comment by mattphilly** [comment]
I also am seeing this issue, but it appears that imshow is running successfully. However, there is a blank popup window and the four lines outputted in the attached image. When I run the test for Gtk and GtkReactive both tests were successful. 

*Running on OSX version Mojave 10.14.5

![Screen Shot 2019-05-16 at 12 17 49 PM](https://user-images.githubusercontent.com/50712051/57870229-abc38380-77d4-11e9-92ad-851d967d34d9.png)

**Comment by vomout** [comment]
I can confirm no image displayed on Mojave, just an empty window.

**Comment by timholy** [comment]
There may be a fix that works for some people in [this specific post](https://github.com/JuliaGraphics/Cairo.jl/issues/271#issuecomment-476827465).

**Comment by timholy** [comment]
Should be fixed if you use Julia 1.3 and up-to-date packages
